File: credit-engine-api/src/main/java/com/credit/engine/application/service/currency/CurrencyService.java
package com.credit.engine.application.service.currency;

import com.credit.engine.application.dto.currency.CurrencyRequest;
import com.credit.engine.application.dto.currency.CurrencyResponse;

import java.util.List;

public interface CurrencyService {

    CurrencyResponse create(CurrencyRequest request);

    CurrencyResponse findByCode(String code);

    List<CurrencyResponse> findAll();

}

File: credit-engine-ui/src/services/currency/currency.service.ts
import { http } from '../../api/client';
import type CurrencyRequest from './dto/CurrencyRequest';
import type CurrencyResponse from './dto/CurrencyResponse';

const controllerUrl = '/currencies';

export const currencyService = {

    findAll() {
        return http.get<CurrencyResponse[]>(controllerUrl);
    },

    findByCode(code: string) {
        return http.get<CurrencyResponse>(`${controllerUrl}/${code}`);
    },

    create(request: CurrencyRequest) {
        return http.post<CurrencyResponse>(controllerUrl, request);
    },

};
